src/System.Runtime.Numerics/tests/ComplexTests.cs

```diff
@@ -353,14 +360,6 @@ public static IEnumerable<object[]> ASin_Advanced_TestData()
             yield return new object[] { -1234000000, 0, -1.5707963267948966, 21.62667394298955 }; // Real part is negative, imaginary part is 0
             yield return new object[] { 0, 1234000000, 0, 21.62667394298955 }; // Imaginary part is positive
 
-            // Boundary values
-            yield return new object[] { double.MaxValue, 0, double.NaN, double.NaN };
```

Comment:
As I mentioned in reply to another comment, I can't really fix these tests without fixing the Asin implementation, because the current Asin implementation is still passing infinities into Sqrt. I have good Asin code which I plan to submit, but it is much more complicated than this simple Sqrt algorithm, so I'd prefer not to further complicate this PR with it.

Here are the options for this PR, please pick one explicitly:
1. For now, simply remove these mathematically wrong asserts. (My preference.) They will be replaced with mathematically correct asserts when Asin is fixed.
2. Replace these mathematically wrong asserts with other mathematically wrong asserts . (Basically, the new Sqrt causes infinities to be returned where before NaNs were returned.) They will be replaced with mathematically correct asserts when Asin is fixed.
3. Replace these mathematically wrong asserts with mathematically correct asserts, and add a bunch of complicated code that replaces the current Asin and Acos implementations to the PR so they are satisfied.
(I am most interested in the view of @mellinoe on this question.)